### ISSUE TITLE
Alpha dropout documentation or implementation is wrong

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#74004]
### 📚 The doc issue

The documentation suggests that "For an input with zero mean and unit standard deviation, the output of Alpha Dropout maintains the original mean and standard deviation of the input"

https://pytorch.org/docs/stable/generated/torch.nn.AlphaDropout.html#torch.nn.AlphaDropout

Let's try this out.

```
import torch
import torch.nn.functional as F

x = torch.randn(100)
x = (x - x.mean()) / x.std()
print(x.mean(), x.std())  # (0, 1)

result = F.alpha_dropout(x, p=0.5, training=True)
print(result.mean(), result.std())  # Not (0, 1)
```

### Suggest a potential alternative/fix

Either the documentation should be updated or the implementation is wrong

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345

**Comment by ngimel** [comment]
It maintains original stats in the same statistical sense regular dropout maintains mean - some expected values for multiple dropout invocations. If you try it on a large tensor, the obtained values will be closer to 0,1

**Comment by albanD** [comment]
Closing as expected behavior.

**Comment by Mypathissional** [comment]
But the formulas seems to be different in here and in the implementation by tensorflow https://github.com/tensorflow/tensorflow/blob/v1.15.0/tensorflow/contrib/nn/python/ops/alpha_dropout.py#L29-L85
